src/lib/mina_base/test/zkapp_precondition_test.ml

```diff
@@ -0,0 +1,104 @@
+(** {1 ClosedInterval tests} *)
+
+let closed_interval_roundtrip_json () =
+  let open Fields_derivers_zkapps.Derivers in
+  let v =
+    { Mina_base.Zkapp_precondition.Closed_interval.lower = 10; upper = 100 }
+  in
+  let full = o () in
+  let _a : _ Unified_input.t =
+    Mina_base.Zkapp_precondition.Closed_interval.deriver ~name:"Int" int full
+  in
+  let result = !(full#of_json) (!(full#to_json) v) in
+  Alcotest.(check bool)
```

Comment:
Nit: check Closed_interval.t instead of check bool? Testable interface should be easy to implement against